cachemonster/services/CacheMonster_ExternalCloudFlareService.php

```diff
@@ -0,0 +1,102 @@
+<?php
+namespace Craft;
+
+/**
+ * Class CacheMonster_ExternalVarnishService
+ *
+ * @package   CacheMonster
+ * @author    Josh Angell
+ * @copyright Copyright (c) 2016, Supercool Ltd
+ * @link      http://plugins.supercooldesign.co.uk
+ */
+
+class CacheMonster_ExternalCloudFlareService extends BaseApplicationComponent implements ICacheMonster_External
+{
+	/**
+	 * @var string
+	 */
+	private static $_cloudFlareApiBaseUrl = 'https://api.cloudflare.com/client/v4/zones/';
+
+	/**
+	 * Purges the given paths
+	 *
+	 * @param array $paths An array of paths to purge
+	 *
+	 * @return bool
+	 */
+	public function purgePaths($paths=array())
+	{
+		if (!$this->_hasValidSettings())
```

Comment:
This is good! I might move that method into the Interface to force other externals to implement it.